deployer/hub.py

```diff
@@ -202,6 +204,53 @@ def auth_aws(self):
                 if orig_kubeconfig is not None:
                     os.environ['AWS_SECRET_ACCESS_KEY'] = orig_secret_access_key
 
+    def auth_azure(self):
+        """
+        Read `azure` nested config, login to Azure with a Service Principal,
+        activate the appropriate subscription, then authenticate against the
+        cluster using `az aks get-credentials`.
+        """
+        config = self.spect['azure']
+        key_path = config['key']
+        cluster = config['cluster']
+        resource_group = config['resource_group']
+
+        with tempfile.NamedTemporaryFile() as kubeconfig:
+            orig_kubeconfig = os.environ.get('KUBECONFIG', None)
+
+            try:
+                os.environ['KUBECONFIG'] = kubeconfig.name
+
+                with decrypt_file(key_path) as decrypted_key_path:
+                    with open(decrypted_key_path) as f:
```

Comment:
On the AWS one, I added this piece:
```python
                    if not os.path.isfile(decrypted_key_abspath):
                        raise FileNotFoundError(
                            'The decrypted key file does not exist')
```
If you feel it makes sense, maybe you can add it here as well...

Reply:
Done!